### PR TITLE
Campus filter

### PR DESCRIPTION
**Pull request by ghost**
# Why 

So that people can more finely search for courses by their location. Alex showerthought: this might be better named "location filter"... is MFA a campus?

# What 

Add a filter for locations. Ie, only show results with the selected locations.

# Spec

Figma Mockup: https://www.figma.com/file/Eh1nMa7vQJ0mxSNIHvyI51/%E2%9C%85-Dev-Ready?node-id=305%3A55

Trello Card: https://trello.com/c/Kd6CfNp5/58-campus-filter

Picture of the Figma mock:

Screen Shot 2020-11-12 at 10 25 44 PM

# Todos

- [x] update db models
- [x] update scrapers
- [x] update backend filters and types
- [x] update frontend filters and types

Daniel's diff: https://github.com/sandboxnu/searchneu/pull/176/files/98cf72f834e2ba042004a280acb976b9bd32278f

### CODE REVIEW

File: backend/searcher.ts [hunk 1]
```diff
@@ -276,21 +275,13 @@ class Searcher {
...
-    const patternResults = query.match(this.COURSE_CODE_PATTERN);
-    const subject = patternResults ? patternResults[1].toUpperCase() : '';
-    // if (patternResults && macros.isNumeric(patternResults[2]) && (this.getSubjects()).has(subject)) {
-    //   ({
-    //     results, resultCount, took, hydrateDuration, aggregations,
-    //   } = await this.getOneSearchResult(subject, patternResults[2], termId));
-    // } else {
     const searchResults = await this.getSearchResults(query, termId, min, max, filters);
-    ({ resultCount, took, aggregations } = searchResults);
+    const { resultCount, took, aggregations } = searchResults;
```
Comment (gamburgm) [inline backend/searcher.ts:281]: gonna have some merge conflicts here, probably want to pair on that or discuss it b/c you're gonna need to update the single search result code.

File: backend/tests/__snapshots__/search.test.seq.js.snap [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`searcher generateMQuery generates aggs with online filters applied 1`] = `
+exports[`searcher generateMQuery generates aggs with campus filters applied 1`] = `
```
Comment (gamburgm) [inline backend/tests/__snapshots__/search.test.seq.js.snap:3]: These snapshot updates are kinda sus, don't know if adding 180 lines of query information should occur just b/c of a changed field in the `sections` schema.

File: backend/tests/updater.test.seq.ts [hunk 3]
```diff
@@ -135,6 +135,7 @@ describe('Updater', () => {
     seatsRemaining: 1,
     waitCapacity: 0,
     waitRemaining: 0,
+    campus: 'Boston',
```
Comment (gamburgm) [inline backend/tests/updater.test.seq.ts]: Can you remove these `campus` lines? They're already being added by the `defaultSectionProps` that you modified on line 102.

File: frontend/components/ResultsPage/Results/DesktopSectionPanel.tsx [hunk 4]
```diff
@@ -72,7 +72,7 @@ function DesktopSectionPanel({ section, showNotificationSwitches } : DesktopSect
           : getMeetings(section)}
       </td>
       <td>
-        Boston
+        {section.campus}
```
Comment (gamburgm) [inline frontend/components/ResultsPage/Results/DesktopSectionPanel.tsx:75]: praise

**Review by gamburgm — COMMENTED** [review]
Overall :chungus: but there are a few issues. One should be quick and easy, one is some merge conflict stuff, and another is a strange snapshot update that we should look into to verify that's the correct output. Overall tho this is great!

File: backend/tests/__snapshots__/search.test.seq.js.snap (hunk 2 above)
Comment (gamburgm) [reply backend/tests/__snapshots__/search.test.seq.js.snap:3]: Nevermind, you're `vibing`

Comment (ghost) [reply backend/tests/__snapshots__/search.test.seq.js.snap:3]: nice haha, i couldn't figure out why they were there lol

File: backend/searcher.ts (hunk 1 above)
Comment (ghost) [reply backend/searcher.ts:281]: yeet

**Pull request closed without merging**